pulsar-client-cpp/lib/ConsumerImpl.cc

```diff
@@ -170,23 +172,25 @@ void ConsumerImpl::connectionOpened(const ClientConnectionPtr& cnx) {
     cnx->registerConsumer(consumerId_, shared_from_this());
 
     Lock lockForMessageId(mutexForMessageId_);
-    Optional<MessageId> firstMessageInQueue = clearReceiveQueue();
-    if (subscriptionMode_ == Commands::SubscriptionModeNonDurable) {
-        // Update startMessageId so that we can discard messages after delivery
-        // restarts
-        startMessageId_ = firstMessageInQueue;
-    }
-    const auto startMessageId = startMessageId_;
+    // Update startMessageId so that we can discard messages after delivery restarts
+    startMessageId_ = clearReceiveQueue();
+    const auto subscribeMessageId = (subscriptionMode_ == Commands::SubscriptionModeNonDurable)
+                                        ? startMessageId_.get()
+                                        : Optional<MessageId>::empty();
     lockForMessageId.unlock();
 
     unAckedMessageTrackerPtr_->clear();
     batchAcknowledgementTracker_.clear();
 
     ClientImplPtr client = client_.lock();
     uint64_t requestId = client->newRequestId();
+    if (duringSeek_) {
+        ackGroupingTrackerPtr_->flushAndClean();
+    }
```

Comment:
It looks like it should be put before `clearReceiveQueue` is called, see https://github.com/apache/pulsar/blob/2848fa0da09e035951220c3d04138041e1477e60/pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java#L820-L831

Reply:
Addressed, but it looks like the order doesn't affect the final result.